File: 240104/hashmap 기본/hashmap-basic.js
let fs = require('fs')
let input = fs.readFileSync('/dev/stdin').toString().trim().split('\n')

const N = parseInt(input[0])
arr = {}
for (let i = 1; i < input.length; i++) {
    const [a, b, c] = input[i].split(" ")
    
    if (a === 'add') {
       arr[b] = c 
    } else if (a === 'remove') {
        delete arr[b]
    } else {
        if (b in arr) {
            console.log(arr[b])
        } else {
            console.log('None')
        }
    }
}

File: 240104/hashmap 기본/hashmap-basic.py
n = int(input())
arr = {}
for _ in range(n):
    line = [i for i in input().split()]
    if line[0] == 'add':
        arr[line[1]] = line[2]
    elif line[0] == 'remove':
        arr.pop(line[1])
    else:
        if line[1] in arr:
            print(arr[line[1]])
        else:
            print("None")